rasa/cli/utils.py

```diff
@@ -233,12 +233,21 @@ def validate_files(
     if stories_only:
         all_good = _validate_story_structure(validator, max_history, fail_on_warnings)
     else:
-        all_good = (
-            _validate_domain(validator)
-            and _validate_nlu(validator, fail_on_warnings)
-            and _validate_story_structure(validator, max_history, fail_on_warnings)
+        if importer.get_domain().is_empty():
```

Comment:
I think we need to expose more explicitly the reasons why a Domain could be empty (via a path.exists() check more appropriate maybe when importer loads the domain?). Also is File not found the only reason a domain could be initialized as empty during validation?

Reply:
I chose this approach since the main issue in the ticket is getting warnings from `rasa data validate`  when the domain is empty (not found or invalid etc.) and not being able to debug why is the pipeline failing while getting domain related warnings. But the wording should be different `Encountered empty domain during validation.` seems descriptive enough 🤔

Reply:
The description still doesn't help the user to find the initial error leading to an empty domain, so that's why I was suggesting best to flag path issues when and where encountered. I'll leave it up to @souvikg10 to decide if best to exit the command if domain path not found, or raise a descriptive warning where Domain gets initialized as empty if path is not found.

Reply:
Hi @Urkem, @ancalita - I think we should exit the command, this is because when in CI, the pipeline to validate can run long, if we don't catch a domain path or incorrect domain path, exiting the command, fails the pipeline immediately so it can be reworked. 
wouldn't this line means, it will exit?
`rasa.shared.utils.cli.print_error_and_exit`

Reply:
@souvikg10 Thanks for chipping in 👍 , yes `rasa.shared.utils.cli.print_error_and_exit` will just exit the `rasa data validate` command with a return value of `1` with the appropriate message without starting any validation.